File: 8kyu/(8 kyu) What is between/(8 kyu) What is between.go
// 1 Plain solution
package kata

func Between(a, b int) []int {
  var res []int
  for i := a; i <= b; i++ {
    res = append(res, i)
  }
  return res
}

// 2 Optimized solution
package kata

func Between(a, b int) []int {
  res := make([]int, b-a+1)

  for i, j := 0, a; j <= b; i, j = i+1, j+1 {
    res[i] = j
  }

  return res
}

// 3 Clever solution
package kata


func Between(a, b int) (res []int) {
  for ; a <= b; a ++ {
    res = append(res, a)
  }
  return
}

File: 8kyu/(8 kyu) What is between/(8 kyu) What is between.rs
// 1 Plain solution
fn between(a: i16, b: i16) -> Vec<i16> {
    let mut result = Vec::new();
    for i in a..=b {
        result.push(i);
    }
    result
}

// 2 Optimized solution
fn between(a: i16, b: i16) -> Vec<i16> {
    (a..=b).map(|x| x).collect()
}

// 3 Clever solution
fn between(a: i16, b: i16) -> Vec<i16> {
    (a..=b).collect()
}

// 4 Coding golf
fn between(a: i16, b: i16) -> Vec<i16> {
    Vec::from_iter(a..=b)
}
